plugins/catalog/src/components/CatalogKindHeader/CatalogKindHeader.tsx

```diff
@@ -59,7 +59,10 @@ export interface CatalogKindHeaderProps {
   initialFilter?: string;
 }
 
-/** @public */
+/**
+ * @public
+ * @deprecated Might be removed in a future release.
```

Comment:
non-blocking comment, but I wonder if this description is good enough. As from the user standpoint this might be sufficiënt to let me know why this is deprecated and if should (not) use it. I'll leave the final judgement to @jhaals

Reply:
Good point @djamaile.
As this deprecation might be controversial for users I suggest that we add some more context and also tell users that want this functionality to remain to create an issue.

For example
```suggestion
* @deprecated This component has been deprecated in favour of the EntityKindPicker in the list of filters. If you wish to keep this component long term make sure to raise an issue at github.com/backstage/backstage
```